### ISSUE TITLE
dup causes a crash

### ISSUE THREAD

**Issue by igorwwwwwwwwwwwwwwwwwwww** [slact/hdr_histogram_ruby#2]
I tried using `hdr.dup` and it caused a crash. It would be cool if there were an implementation of `dup` which safely copies the histogram.

I was able to work around this via:

```
hdr2 = HDRHistogram.new(...)
hdr2.merge(hdr)
```

**Comment by slact** [comment]
> It would be cool if it wouldn't crash.

It sure would. However, I'm unable to reproduce this on ruby 2.5, nor 2.4 . Can you give me your ruby version and some code that crashes on `dup` for you?

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
oh totally, sorry for now including that. :)

reproduce case:

```ruby
require 'HDRHistogram'

hdr = HDRHistogram.new(1, 3600, 3)
hdr.dup.percentile(99)
```

was able to reproduce on:

```
➜  ruby --version
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin16]
```

and also:

```
➜  ruby --version
ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin17]
```

**Comment by slact** [comment]
Confirmed, thanks. will update when it's fixed.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Is there any update here? I saw there is now a `hdr.merge!` method?

**Comment by slact** [comment]
The bad news is that I completely forgot about this issue. The good news is that I just fixed it. Give v 0.1.8 a try.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
I tried it out, one issue is that from what I can tell, the unit and multiplier are not copied as part of the dup.

**Comment by slact** [comment]
aaand fixed in 0.1.9

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Confirmed working! Thanks!
